### PR TITLE
actions: add edit draft

### PR DESCRIPTION
**Pull request by ppanero**
- partly closes https://github.com/zenodo/zenodo-rdm/issues/428

### CODE REVIEW

File: invenio_rdm_migrator/streams/actions/load/drafts.py [hunk 1]
```diff
@@ -188,3 +187,54 @@ def _resolve_references(self, **kwargs):
         if communities:
             self.resolve_communities(communities)
         self.resolve_draft_pids(self.data.draft)
+
+
+@dataclass
+class RDMDraftEditData(LoadData):
+    """Draft edit action data."""
```
Comment (slint) [inline invenio_rdm_migrator/streams/actions/load/drafts.py:194]: question: is this action handling both the `POST /api/deposit/depositions/{id}/actions/edit` and the `PUT /api/deposit/depositions/{id}`? These are separate individual actions in legacy (and in RDM AFAIK). If they're both handled by this class this is fine though.

**Review by slint — APPROVED** [review]
LGTM! Once comment on which action exactly does this map to

File: invenio_rdm_migrator/streams/actions/load/drafts.py (hunk 1 above)
Comment (ppanero) [reply invenio_rdm_migrator/streams/actions/load/drafts.py:194]: The payloads where gathered when interacting with the UI, so the target API endpoint is `PUT /api/deposit/depositions/123456`. What are the differences from both of the above?

Comment (ppanero) [reply invenio_rdm_migrator/streams/actions/load/drafts.py:194]: As per comment here https://github.com/zenodo/zenodo-rdm/pull/452#discussion_r1296859595, this PR is ready to merge

Comment (slint) [reply invenio_rdm_migrator/streams/actions/load/drafts.py:194]: There are 3 ways to "edit" a deposit:

Type | Action | Entryopint | URL
-- | -- | -- | --
Record/draft | Edit record/deposit | REST API | `POST /api/deposit/depositions/{id}/actions/edit`
Record/draft | Edit record/deposit | UI | `GET /deposit/{id}` ()
Record/draft | Edit record/deposit | UI | `POST /record/{id}` -> (redirects to `GET /deposit/{id}`

In the end they probably all produce the same transaction/statement (at least the 2 first ones). The `GET /deposit/{id}` [conditionally does so here](https://github.com/zenodo/zenodo/blob/daf297d580b9f40f7497825d77ba0797903e48a5/zenodo/modules/deposit/views.py#L392-L395).

Comment (ppanero) [reply invenio_rdm_migrator/streams/actions/load/drafts.py:194]: Question isn't the second case: a `PUT` op and the sentence if `in draft state`...? A `GET` seems somehow a way of getting a new deposit. (but that happens with `POST`).

Sorry, I'm trying to understand, but I'm getting confused I lack quite some experience with legacy.

Comment (slint) [reply invenio_rdm_migrator/streams/actions/load/drafts.py:194]: In legacy, the `GET /deposit/{id}` sneaks in an "edit" to the draft if it's not already in "edit"/"draft" state. REST API clients can't do this though, they have to go the "proper way" and first do their `POST /api/deposit/depositiosn/{id}/actions/edit` call.

**Pull request merged**